packages/babel-types/src/clone/cloneNode.js

```diff
@@ -0,0 +1,35 @@
+import { NODE_FIELDS } from "../definitions";
+
+const has = Function.call.bind(Object.prototype.hasOwnProperty);
+const isNode = obj => obj && has(NODE_FIELDS, obj.type);
+
+export default function cloneNode<T: Object>(node: T): T {
+  if (!node) return node;
+
+  // Special-case identifiers since they are the most cloned nodes.
+  if (node.type === "Identifier") {
```

Comment:
Maybe better to save `type` to a local var (`const { type } = node`) since you reuse it at least 3 times